### ISSUE TITLE
Docker-in-Docker GitLab CI

### ISSUE THREAD

**Issue by netzdoktor** [cross-rs/cross#513]
I am using `cross` in a docker-in-docker scenario on GitLab CI. I used the approach given in https://github.com/rust-embedded/cross/issues/273#issuecomment-528300479, which unfortunately did not work anymore. I am still getting:

```
sh: 1: cargo: not found
```

So I used `CROSS_DOCKER_IN_DOCKER=true` (as recommended in another comment) and the result is this:

```
error: `"/usr/bin/docker" "inspect" "runner-27oc64mp-project-587-concurrent-0"` failed with exit code: Some(1)
```

I have the latest release of cross installed (but checking a diff with `master` didn't reveal relevant changes as far as I can tell).

I am wondering why this does not work in the GitLab environment...

Thanks in advance!

**Comment by gaoyifan** [comment]
Gitlab runner set a different hostname, which is not equal to conatiner name. Like this:

```
container name:  runner-dtfmkqbc-project-33-concurrent-0-22fbe3329596e791-build-2
hostname: runner-dtfmkqbc-project-33-concurrent-0
```

In cross, we assume that hostname is equal to the container name:
https://github.com/rust-embedded/cross/blob/4f213ba07b214d36225be63e5b72a6346937a699/src/docker.rs#L237

I have not find a solution yet.

**Comment by gaoyifan** [comment]
Workaround: 
```yaml
  script:
    - export HOSTNAME=$(docker ps -ql)
    - cross build ...
```

**Comment by Kaiser1989** [comment]
> Workaround:
> 
> ```yaml
>   script:
>     - export HOSTNAME=$(docker ps -ql)
>     - cross build ...
> ```

Calling `$(docker ps -ql)` returns empty string in my case. Are there any other possibilities to get correct container name?

**Comment by Alexhuszagh** [comment]
> 

That suggests there are no created containers, of any state (running or stopped). Try using `docker ps -a` to print all running containers, which I expect would be empty.

**Comment by Alexhuszagh** [comment]
This is a duplicate of #273. We've documented how to use [docker-in-docker](https://github.com/cross-rs/cross/wiki/FAQ#docker-in-docker), [remote docker](https://github.com/cross-rs/cross/wiki/Remote) (required if Gitlab is using a remote Docker connection, such as with a host at `tcp://docker:2375/`).

**Comment by sybernomad** [comment]
> Workaround:
> 
> ```yaml
>   script:
>     - export HOSTNAME=$(docker ps -ql)
>     - cross build ...
> ```

This didn't work for me when I had multiple cross builds running in parallel. I got around it by doing this instead:

```yaml
   script:
     - export HOSTNAME=$(docker ps -aqf "name=^$HOSTNAME" | awk 'NR==1{print $1}')
     - cross build ...
```
